### ISSUE TITLE
"Bind Events" Not Firing

### ISSUE THREAD

**Issue by kabassmusic** [AgoraIO-Community/Agora-Unreal-SDK-Blueprint#5]
This is the THIRD time I have typed this out in a week. I have been transferred by Agora support twice already. 

I have setup and prepared the project as the official tutorials have guided me. Unfortunately ALL of the "Bind Event" nodes are not working. Everything else is working. I have even setup "Print String" nodes through everything. My App ID is correct and there are no other issues. 

I have even tested this in the demo project as well with no success. 

Unreal Engine 4.25

**Comment by plutoless** [comment]
@kabassmusic sorry for the inconvenience caused. we will release a new version in Jan 15th. it will be a fully tested version and adapted to latest sdk. maybe you can try on that directly.

**Comment by kabassmusic** [comment]
Okay. Thank you for your quick response. I look forward to the next update. This seems very powerful for what I'm trying to accomplish.

**Comment by joelthomas362** [comment]
@kabassmusic Can you please post a picture of the code where your nodes aren't working? Thanks!

**Comment by kabassmusic** [comment]
I have covered my AppID for protection. 
When I run this, I get:
- AGORA INITIALIZED
- 0
- 0
- NOTHING

![GameInstance](https://user-images.githubusercontent.com/28176852/104038416-b5181580-519a-11eb-981d-eb2e693f537a.png)
![ChannelJoiner](https://user-images.githubusercontent.com/28176852/104038431-b8ab9c80-519a-11eb-8489-28bf52bbdd11.png)
![AgoraManager_001](https://user-images.githubusercontent.com/28176852/104038441-bc3f2380-519a-11eb-8154-743430ffc84a.png)
![AgoraManager_002](https://user-images.githubusercontent.com/28176852/104038449-be08e700-519a-11eb-9691-36bed7506528.png)
![AgoraManager_003](https://user-images.githubusercontent.com/28176852/104038455-c06b4100-519a-11eb-825d-aa8b4e94cbfc.png)

**Comment by kabassmusic** [comment]
If I do this, then the output properly says "LOCAL". 
Also, the "Agora Video Setup" node does the same thing and won't fire the "Bind Event" nodes. 
![AgoraManager_004](https://user-images.githubusercontent.com/28176852/104040090-d7ab2e00-519c-11eb-9b64-51fb0cad01af.png)

**Comment by joelthomas362** [comment]
From the "Agora Engine" node, type in 'Assign on user joined delegate'. 
Doing so will create two nodes, the Bind Event and the Callback event to which you plug in the functionality you want. It looks like this:
![image](https://user-images.githubusercontent.com/10685671/104219891-b781a600-540c-11eb-9b27-3c4d1e24c958.png)

**Comment by kabassmusic** [comment]
Yeah that's how I set it up prior to this conversation. Tried again and I still have the same results. 
My screenshots just look different to you because I renamed them. But even without renaming, It still occurs. 

EDIT: The Echo Test in the demo works with the "On Join Channel Success Delegate". I placed it inside of the "VideoCallWidget_BP" at the end of the "Event Construct" and that's the only time "Join Channel Success" has worked so far. That's a good sign at least.

EDIT 2: When testing the camera, I get Warning codes 8 and 16. But the camera feed is showing up in the preview. 
I also get this warning when opening the project. 

![agorawarning](https://user-images.githubusercontent.com/28176852/104285105-b9814e80-5478-11eb-837e-acd4602f0870.png)

**Comment by kabassmusic** [comment]
So I tried rebuilding the visual studio files. Nothing changed. 

Then I hit compile in Unreal Engine and it broke the whole plugin. :(

**Comment by plutoless** [comment]
@kabassmusic 
sorry for the delay, but we are glad to let you know that we have released our stable community unity sdk version at
https://github.com/AgoraIO-Community/Agora-Unreal-SDK/
the delivery is bit late due to some technical issues. we are looking forward to receiving your feedback.

**Comment by kabassmusic** [comment]
@plutoless 

So I have finally gotten around to testing out the new version but it wouldn't open or auto build for me.

Then opened it inside of VS and tried to build it, I got these errors. 
![cant-build](https://user-images.githubusercontent.com/28176852/114289620-0efcd580-9a3f-11eb-9515-c4f98c19a1d6.png)

**Comment by zhangtao1104** [comment]
I think maybe you didn’t download the new sdk.
 You need to run the new demo according to the readme.

**Comment by kabassmusic** [comment]
Ah yeah. That was my fault. 

I was doing this in my Downloads folder. 
It wasn’t allowing me to copy all of the files from the SDK due to file names being too long. 


- But then I redownloaded the project and new SDK. Copied the AgoraPlugin folder to the Plugins folder in the API_Example. 
- Right clicked the project file, Generated VS files, and launched the project. 
- The project needed to be rebuilt, so I clicked okay, build fails. 
- Then I opened the .sln file in VS, Build from there, and it also fails with dozens of errors.

**Comment by zhangtao1104** [comment]
I had try the project. I can build it successful. I build it by Unreal 4.26.
Your error log is 'Can't open include file ''UVideoTextureObserver.h': No such file or directory. " ?

The UVideoTextureObserver.h is part of the AgoraPlugin, you can try to find it in AgoraPlugin/Source/AgoraPlugin/Public/ UVideoTextureObserver.h.


You can check if you run it as these steps:
1: download AgoraPlugin unzip it and copy it into API_Example/Plugins/.
2: Fill in your appId into the the file  API_Example/Source/Agora_Unreal_Video/AgoraController.h.
3: Double click the API_Example.uproject by  Unreal.

**Comment by kabassmusic** [comment]
I got past that error in my last comment. But now I was getting dozens of other errors after doing it properly. 

I'm using 4.25.2. 

Would I get these extra errors because I didnt fill out the appid?

**Comment by zhangtao1104** [comment]
Yes, If you didn't fill in appId, you will get other error, you need to fill in the appId and build it. Can you doneload 4.26 and have a try. I will download 4.25 and have a try.

**Comment by kabassmusic** [comment]
Okay. Yeah I'll try that out sometime this week and report back here when I'm done.

**Comment by zhangtao1104** [comment]
@kabassmusic I had try 4.25.4, I can run successful. I think the reason why you still get error is that you didn't fill in the appId.
Y

**Comment by kabassmusic** [comment]
Alright. After I have entered my appid with quotations, it finally opened in Unreal. Thank you for your help. 

Is there any new tutorials for this version?

**Comment by zhangtao1104** [comment]
Ok, You need to fill in the right AppId, or you can't run the voice call and video call successful.

https://github.com/AgoraIO-Community/Agora-Unreal-SDK  This Project is our newest project, it contains a voice sample and a video sample. We will keep update this project, you just need to follow that project.
